lib/services/local/users.go

```diff
@@ -287,10 +328,11 @@ func (s *IdentityService) CompareAndSwapUser(ctx context.Context, new, existing
 		return trace.Wrap(err)
 	}
 	existingItem := backend.Item{
-		Key:     backend.Key(webPrefix, usersPrefix, existing.GetName(), paramsPrefix),
-		Value:   existingValue,
-		Expires: existing.Expiry(),
-		ID:      existing.GetResourceID(),
+		Key:      backend.Key(webPrefix, usersPrefix, existing.GetName(), paramsPrefix),
+		Value:    existingValue,
+		Expires:  existing.Expiry(),
+		ID:       existing.GetResourceID(),
+		Revision: existing.GetRevision(),
```

Comment:
I can understand wanting to be consistent in the other functions (although we only need to pass along revision and ID on upsert, for the sake of the cache) but for the existing item on a CAS there's really no reason.
```suggestion
```